### DISCUSSION TITLE
Provide a mechanism for customizing build host configuration

### DISCUSSION THREAD

**Discussion by tammymakesthings** [blinksh/blink#1793] [Ideas]
It would be great if there was a way to customize the configuration of my build host any time it’s (re-)created. Ideally, the customization would be in the form of a shell script that could do things like:

- Install additional software packages
- Change the user’s shell
- Sync files with a remote host
- etc.

I know I can sort of kludge at least some of this by editing my `~/.bash_login` file, but it would be nice if there was a supported way to override/customize the configuration.

**Comment by carloscabanero** [comment]
Thanks a lot for taking the time to test Build and the suggestion!

We were actually thinking about a script that could run after starting a specific container (say for example, a mysql.sh script or a ht.sh for hacker tools). That way we could initialize, finalize setups, etc... For example:
- We could start applications in the background, like VS Code.
- We could start compilations, or download a cache, etc... for very large projects. Probably syncing that needs to happen in a specific machine (ie, downloading data sets?)
- Change the default user shell could be done here as well.
- Installing additional software packages that may not be in a specific container and that you don't want added in a custom container for some reason would also fit here.

For the whole machine, this may make sense as well:
- Maybe pre-loading other containers that you usually run.
- Maybe downloading other disks or additional data you may want to have on a specific container.
- Other ideas?

We are still testing the waters, so ideas like these are greatly appreciated. We will then go back and do another big push before we take this out of Beta.
